opm/material/fluidmatrixinteractions/TwoPhaseLETCurves.hpp

```
@@ -0,0 +1,312 @@
+ *
+ *\see LETCurvesParams
+ */
+template <class TraitsT, class ParamsT = TwoPhaseLETCurvesParams<TraitsT> >
+class TwoPhaseLETCurves : public TraitsT
+{
+public:
+    typedef TraitsT Traits;
+    typedef ParamsT Params;
+    typedef typename Traits::Scalar Scalar;
+
+    static_assert(Traits::numPhases == 2,
+                  "The number of fluid phases must be two if you want to use "
+                  "this material law!");
+
+    static constexpr Scalar eps = 1.0e-10; //tolerance
+
+    //! The number of fluid phases to which this material law applies.
+    static const int numPhases = Traits::numPhases;
+
+    //! Specify whether this material law implements the two-phase
+    //! convenience API
+    static const bool implementsTwoPhaseApi = true;
+
+    //! Specify whether this material law implements the two-phase
+    //! convenience API which only depends on the phase saturations
+    static const bool implementsTwoPhaseSatApi = true;
+
+    //! Specify whether the quantities defined by this material law
+    //! are saturation dependent
+    static const bool isSaturationDependent = true;
+
+    //! Specify whether the quantities defined by this material law
+    //! are dependent on the absolute pressure
+    static const bool isPressureDependent = false;
+
+    //! Specify whether the quantities defined by this material law
+    //! are temperature dependent
+    static const bool isTemperatureDependent = false;
+
+    //! Specify whether the quantities defined by this material law
+    //! are dependent on the phase composition
+    static const bool isCompositionDependent = false;
+
+    /*!
+     * \brief The capillary pressure-saturation curves.
+     */
+    template <class Container, class FluidState>
+    static void capillaryPressures(Container& /* values */, const Params& /* params */, const FluidState& /* fs */)
+    {
+        throw std::invalid_argument("The capillaryPressures(fs) method is not yet implemented");
```

All of these should throw std::logic_error instead of invalid_argument.

---

Fixed.